commons/src/test/java/io/aiven/kafka/tieredstorage/commons/SegmentCompressionCheckerTest.java

```diff
@@ -0,0 +1,184 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Writer;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+
+import java.nio.file.StandardOpenOption;
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.FileRecords;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.common.record.MemoryRecordsBuilder;
+import org.apache.kafka.common.record.TimestampType;
+
+import org.assertj.core.util.Files;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+class SegmentCompressionCheckerTest {
+
+    @TempDir
+    Path dir;
+
+    @Test
+    void shouldFailWhenReadingEmptyFile() {
+        try {
+            SegmentCompressionChecker.check(Files.newTemporaryFile());
+            fail("Expected to fail");
+        } catch (InvalidRecordBatchException e) {
+            assertThat(e).hasMessage("Record batch is null");
+        }
+    }
+
+    @Test
+    void shouldFailWhenReadingInvalidFile() throws IOException {
+        // Given a valid record batch
+        final Path path = dir.resolve("invalid.log");
+        final File file = Files.newFile(path.toString());
+        try (final FileRecords records = FileRecords.open(file, false, 100000, true)) {
+            try (final MemoryRecordsBuilder builder = MemoryRecords.builder(
+                ByteBuffer.allocate(1024),
+                CompressionType.NONE,
+                TimestampType.CREATE_TIME,
+                0)) {
+                builder.append(0L, "key-0".getBytes(), "value-0".getBytes());
+                records.append(builder.build());
+            }
+        }
+        // When messing with the record content
+        final byte[] bytes = java.nio.file.Files.readAllBytes(path);
+        final byte[] output = new byte[bytes.length];
+        for (int i = 0; i < bytes.length; i++) {
+            if (bytes[i] != "0".getBytes()[0]) {
+                output[i] = bytes[i];
+            } else {
+                output[i] = "1".getBytes()[0];
+            }
+        }
+        java.nio.file.Files.write(path, output);
+        // Then
+        try {
+            SegmentCompressionChecker.check(file);
+            fail("Expected to fail");
+        } catch (InvalidRecordBatchException e) {
+            assertThat(e).hasMessage("Failed to read and validate first batch");
+        }
```

Comment:
I think it should be something like
```suggestion
        assertThatThrownBy(() -> SegmentCompressionChecker.check(file))
            .isInstanceOf(InvalidRecordBatchException.class)
            .hasMessage("Failed to read and validate first batch");
```